File: ios-swift/Cronwatch/AppEnvironment.swift
import Foundation

enum AppEnvironment {
    static var firebaseAPIKey: String?              { value(for: "FIREBASE_API_KEY") }
    static var firebaseProjectID: String?           { value(for: "FIREBASE_PROJECT_ID") }
    static var firebaseAppID: String?               { value(for: "FIREBASE_APP_ID") }
    static var firebaseSenderID: String?            { value(for: "FIREBASE_MESSAGING_SENDER_ID") }
    static var firebaseStorageBucket: String?       { value(for: "FIREBASE_STORAGE_BUCKET") }
    static var googleIOSClientID: String?           { value(for: "GOOGLE_IOS_CLIENT_ID") }
    static var captureProxyURL: URL?                { value(for: "CAPTURE_PROXY_URL").flatMap(URL.init(string:)) }
    static var revenueCatAPIKey: String?            { value(for: "REVENUECAT_API_KEY_IOS") }

    private static func value(for key: String) -> String? {
        if let s = Bundle.main.object(forInfoDictionaryKey: key) as? String, !s.isEmpty, !s.hasPrefix("$(") { return s }
        if let s = ProcessInfo.processInfo.environment[key], !s.isEmpty { return s }
        return nil
    }
}

File: android/app/src/main/java/app/cronwatch/AppEnvironment.kt
package app.cronwatch

object AppEnvironment {
    val firebaseProjectId: String? get() = BuildConfig.FIREBASE_PROJECT_ID.ifBlankNull()
    val firebaseApiKey: String? get() = BuildConfig.FIREBASE_API_KEY.ifBlankNull()
    val firebaseAppId: String? get() = BuildConfig.FIREBASE_APP_ID.ifBlankNull()
    val firebaseSenderId: String? get() = BuildConfig.FIREBASE_SENDER_ID.ifBlankNull()
    val firebaseStorageBucket: String? get() = BuildConfig.FIREBASE_STORAGE_BUCKET.ifBlankNull()
    val googleWebClientId: String? get() = BuildConfig.GOOGLE_WEB_CLIENT_ID.ifBlankNull()
    val captureProxyUrl: String? get() = BuildConfig.CAPTURE_PROXY_URL.ifBlankNull()
    val revenueCatApiKey: String? get() = BuildConfig.REVENUECAT_API_KEY_ANDROID.ifBlankNull()

    private fun String.ifBlankNull(): String? = takeIf { it.isNotBlank() }
}
